File: src/main/java/com/vicen/films/framework/presentation/rest/dto/StaffMemberDtoOut.java
package com.vicen.films.framework.presentation.rest.dto;

import com.vicen.films.domain.Identifiable;
import com.vicen.films.domain.people.StaffMember;

import java.util.Objects;
import java.util.Optional;

public class StaffMemberDtoOut extends StaffMemberDtoIn {

    private String id;

    private StaffMemberDtoOut(String id, String name, String lastName, String birthDate) {
        this.id = id;
        this.name = name;
        this.lastName = lastName;
        this.birthDate = birthDate;
    }

    public static StaffMemberDtoOut of(StaffMember staffMember) {
        if (Objects.isNull(staffMember)) {
            return null;
        }
        return new StaffMemberDtoOut(
                Identifiable.getIdFrom(staffMember).toString(),
                staffMember.getName(),
                staffMember.getLastName(),
                Optional.ofNullable(staffMember.getBirthDate()).map(Objects::toString).orElse(null));
    }

    public String getId() {
        return id;
    }

}

File: src/main/kotlin/com/vicen/films/framework/presentation/rest/dto/StaffMemberDtoOut.kt
package com.vicen.films.framework.presentation.rest.dto

import com.vicen.films.domain.Identifiable
import com.vicen.films.domain.people.StaffMember

class StaffMemberDtoOut() : StaffMemberDtoIn() {

    var id: String? = null

    constructor(staffMember: StaffMember) : this() {
        id =  Identifiable.getIdFrom(staffMember).toString()
        name = staffMember.name
        lastName = staffMember.lastName
        birthDate = staffMember.birthDate.toString()
    }

}